src/interp.rs

```diff
@@ -19,153 +19,483 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-use smallvec::SmallVec;
-use std::f64;
-
-/// Data structure for polyphase FIR interpolator
+#[allow(clippy::large_enum_variant)]
 #[derive(Debug)]
-pub struct Interp {
-    /// Interpolation factor of the interpolator
-    factor: usize,
-    /// Taps (prefer odd to increase zero coeffs)
-    taps: usize,
-    /// Number of channels
-    channels: u32,
-    /// Size of delay buffer
-    delay: usize,
-    /// List of subfilters (one for each factor)
-    filter: SmallVec<[Filter; 4]>,
-    /// List of delay buffers (one for each channel)
-    z: Box<[f32]>,
-    /// Current delay buffer index
-    zi: usize,
+enum _Interp {
+    Mono4F(specialized::Interp4F<[f32; 1]>),
+    Stereo4F(specialized::Interp4F<[f32; 2]>),
+    Surround4F(specialized::Interp4F<[f32; 6]>),
```

Comment:
OOC, Why not a 4 channel variant? Simply because it's very uncommon (L, R, C and LFE I guess?) compared to everything else?

Reply:
I considered it not common enough to warrant its own specialization. I've read that for a small number of variations, match-dispatching on a small enum is faster than dynamic dispatch `Box<dyn Interpolator>`, but I'm not sure where the limit is. Perhaps worth to explore in a later PR? I'd really also want 8ch (Front, Mid and Rear + Center and LFE) to be accelerated but left that out too to avoid enum-overhead.

Reply:
After looking into "dyn" dispatch as well. Seems to not be significantly slower than the 7 option enum-dispatch, so I'm adding QuadFrames and Surround8Frames.

Reply:
Is there any difference performance-wise between the big enum and the trait object with dynamic dispatch?

Reply:
Doesn't seem like it, I benched before & after and got similar results. The dispatch is not in a very hot loop anyways, so I shouldn't be surprised. I've replaced the enum with dynamic dispatch, since I think it's a bit "cleaner" given it does not affect performance.

Reply:
Yeah doesn't really matter to me either way if it doesn't affect performance and this is cleaner indeed. Sometimes these things affect performance in surprising ways because here you now have another pointer indirection but as you say... :)